### PR TITLE
Associate profile to logged in user 

### LINKED ISSUE #7065: [FEATURE] Associate profile to logged in user
### Description

>  if username is different from logged in user, show a warning and have a button to update (this will only work if someone logged in before they changed their username)

USER
- [ ] When user logs in associate the profile to their account using their GitHub username (maybe GitHub ID?)

Ideas...
- I think the model `models/Account.js` needs to be updated to create a on-to-many relationship to `models/Profile.js` on login (or on profile update in PR #6998?) 
- then the queries in the app should not check `Profile` model for username but use the relationship of the logged in user
- this will then help us resolve the issue of people changing their GitHub username and also allow people to create multiple profiles (in the future)

These 2 have been moved to their own issue https://github.com/EddieHubCommunity/LinkFree/issues/7349
- [ ] ~~If the username no longer matches the logged in user, show an error of their previous associated username and their current GitHub username~~
- [ ] ~~When displaying the error, have a button to update their username and their url will also change~~

~~Dependent on https://github.com/EddieHubCommunity/LinkFree/issues/7066~~

### Screenshots

_No response_

### Additional information

After this issue is complete, then this issue needs to be done https://github.com/EddieHubCommunity/LinkFree/issues/7349

### PR DESCRIPTION
**Pull request by raviraj-github**
## Fixes Issue

Closes #7065

## Changes proposed
- [x]  - The account model has been extended to facilitate a one-to-many relationship with profiles.
- [x]  - An account API has been implemented to retrieve account information from the database and establish profile associations during the initial user login.
- [x]  - Statistics.js has been updated to call the account API.

## Check List (Check all the applicable boxes) 

- [x] My code follows the code style of this project.
- [ ] My change requires changes to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] All new and existing tests passed.
- [x] This PR does not contain plagiarized content.
- [x] The title of my pull request is a short description of the requested changes.

## Screenshots

image

## Note to reviewers

I have used info log for account js but that could be changed to debug once the logging issue PR merged. 

I have also fixed the comment given by all reviewers in PR https://github.com/EddieHubCommunity/LinkFree/pull/7721

<a href="https://gitpod.io/#https://github.com/EddieHubCommunity/LinkFree/pull/8232"></a>

### CODE REVIEW

File: pages/api/account/account.js [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+import { getServerSession } from "next-auth/next";
+import connectMongo from "@config/mongo";
+import { Account } from "@models/index";
+
+export default async function handler(req, res) {
+  const session = await getServerSession(req, res, authOptions);
+
+  if (!session) {
+    res.status(401).json({ message: "You must be logged in." });
+    return;
```
Comment (eddiejaoude) [inline pages/api/account/account.js:12]: I like the style on line 16, where the return is on the same line as the `res` - I think we can use that style in other places of the code 👍 

```suggestion
    return res.status(401).json({ message: "You must be logged in." });
```

File: pages/account/statistics.js [hunk 2]
```diff
@@ -31,9 +32,17 @@ export async function getServerSideProps(context) {
...
+  let account = await getAccountByProviderAccountId(session.user.id);
+
   const username = session.username;
   const { status, profile } = await getUserApi(req, res, username);
-  if (status !== 200) {
+
+  if (status === 200) {
+    //Linking a profile to an account is a one-time process. Subsequent logins will disregard any attempt to add the profile again.
+    if (account) {
+      await associateProfileWithAccountIfAbsentAndReturnAccount(account, profile._id);
```
Comment (eddiejaoude) [inline pages/account/statistics.js:43]: Thinking about it, could this be triggered on login instead of the stats page?

**Comment by eddiejaoude** [comment]
Looks good 👍 I am just going to experiment with a few things

File: pages/api/account/account.js [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+
+  if (req.method !== "GET") {
+    return res.status(400).json({ error: "Invalid request: GET request required" });
+  }
+
+  try {
+    const data = await getAccountByProviderAccountId(session.user.id);
+    res.status(200).json(data);
+  } catch (error) {
+    logger.error('Error in handler:', error);
```
Comment (eddiejaoude) [inline pages/api/account/account.js:23]: The exception is the first parameter...

```suggestion
    logger.error(error, 'Error in Account handler');
```

**Comment by eddiejaoude** [comment]
Merged to temporary branch and making tweaks in PR https://github.com/EddieHubCommunity/LinkFree/pull/8315

File: pages/account/statistics.js (hunk 2 above)
Comment (raviraj-github) [reply pages/account/statistics.js:43]: Let me come back on this.

File: pages/api/account/account.js (hunk 1 above)
Comment (raviraj-github) [reply pages/api/account/account.js:12]: Sure. But I think now you have merged the PR. Should I raise a new commit?

File: pages/account/statistics.js (hunk 2 above)
Comment (eddiejaoude) [reply pages/account/statistics.js:43]: After playing I managed to learn more about next-auth and do this when signing in, so I merged your PR into a temporary branch and  made my changes, you can see both our changes here https://github.com/EddieHubCommunity/LinkFree/pull/8315

File: pages/api/account/account.js (hunk 1 above)
Comment (eddiejaoude) [reply pages/api/account/account.js:12]: no problem, I updated already, it was just info for next time

**Pull request merged**